src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -587,31 +587,42 @@ def get_refresh_token(self, resource=None,
         sp_secret = self._creds_cache.retrieve_secret_of_service_principal(username_or_sp_id)
         return username_or_sp_id, sp_secret, None, str(account[_TENANT_ID])
 
-    def get_raw_token(self, resource=None, subscription=None):
+    def get_raw_token(self, resource=None, subscription=None, tenant=None):
+        if subscription and tenant:
+            raise CLIError("Please specify only one of subscription and tenant, not both")
         account = self.get_subscription(subscription)
```

Comment:
if subscription is not specified, account is empty? then can we get the user_type below?

Reply:
If subscription is not specified, it retrieves the `default` subscription which is the one that is currently selected by `az account set`.